### ISSUE TITLE
Some tabs randomly stop working entirely

### ISSUE THREAD

**Issue by mcmerdith** [zen-browser/desktop#3102]
### Captchas

- [X] I have read the instructions.
- [X] I have searched existing issues and avoided creating duplicates.
- [X] I am not filing an enhancement request.

### What happened?

Some tabs randomly "die". They just go blank and do nothing, they can't even open devtools. I've attached a ~1 minute video of it happening on my computer. Potentially due to tabs becoming inactive as it's never happened to an active tab, but I honestly have no idea what causes it. I didn't include it in the video but going to a different site doesn't change anything. Still dead afterwards. Probably irrelevant but I ran a profiler on the tab while refreshing it and it still seems to do things, it just doesn't display them.

I couldn't find any logs, but perhaps I'm not looking in the right place. If you need logs just tell me where to find them.

I couldn't reproduce it in firefox, but it's such an occasional issue that to have it happen I would have to try for at least a couple hours.

[Zen 2024-11-19 01.45 profile.json.gz](https://github.com/user-attachments/files/17810787/Zen.2024-11-19.01.45.profile.json.gz)

https://github.com/user-attachments/assets/9dbb7d17-172f-48ff-9acc-1d21f338c23d

Thanks :)

### Reproducible?

- [X] I have checked that this issue cannot be reproduced on Mozilla Firefox.

### Version

1.0.1-a.19

### What platform are you seeing the problem on?

Windows

**Comment by strayobject** [comment]
This happens on linux as well. Clicking on the tabs that hang up does not bring them into focus. The only thing I can do is close them.

**Comment by multimokia** [comment]
Yeah I've noticed this too, often having to duplicate the tab to get it back, but that often yields a lot of lost work. I've noticed this tends to happen with _unloaded tabs_ with a machine that's struggling to have enough RAM all around, so I'm running a test without the unloader to see if things are any better. I'll report back if I'm still experiencing the issue.

On my main PC at home, I have not experienced this, neither on Windows or Linux, but I also have 64 Gb of RAM.

**Comment by mcmerdith** [comment]
Just happened to me 3 times on my laptop (which is much lower spec'd). I Ctrl+Click opened 3 tabs which opened in the little preview window, then I expanded them to their own tabs, but when I actually clicked in 2/3 of them were dead

**Comment by multimokia** [comment]
Update, I can confirm it _is_ the tab unloader. I have not had any issues since disabling it.

That said I think I've noticed a second issue with the tab unloader -- any devtools instance attached to a tab will get disconnected if the tab unloads, which has been immensely frustrating.

I might just disable it outright as a result. I kind of wish the unloader could be a little smarter in how it handles automatic unloading vs a simple timer, likewise that I could unload on demand only, or set specific regexes to never unload. (Office online is a big problem for this as unloaded tabs will always refresh the document losing the place you were at last)

**Comment by iamyuu** [comment]
This happening on M1 pro too, after the tab hang / freeze I even can't click the tabs (just close). I think this is critical issues, should be fixed as soon as possible.

**Comment by mr-cheffy** [comment]
Does this still happen?

**Comment by iamyuu** [comment]
yes, still happening on my device

**Comment by iamyuu** [comment]
Unfortunately, still hapening on my device, tabs still crash. In my case, I open [noteshub.app](https://www.noteshub.app/) and edit the note, then I leave the tab and do something else (browse, etc), when I try to go back to the tab, the tab doesn't open (I've already tried to open it using the keyboard or mouse)

**Comment by mr-cheffy** [comment]
## Closing this issue as part of a repo cleanup

Due to technical-debt we had in the past, we are currently clearing out old, duplicate, stale, and non-actionable issues to improve issue tracking and make it easier to focus on fixing real bugs.

If this issue is still relevant, please open a new issue using our updated issue template, which will help us categorize and address it more efficiently.

Thanks for your understanding and for helping us improve the project! 🚀

This issue can be reopened at: https://github.com/zen-browser/desktop/issues/new?template=bug_report.yml&title=Some%20tabs%20randomly%20stop%20working%20entirely%0A
